pywbem/cim_operations.py

```diff
@@ -1172,12 +1209,230 @@ def __repr__(self):
                "default_namespace=%r, x509=%s, verify_callback=%r, " \
                "ca_certs=%r, no_verification=%r, timeout=%r, " \
                "use_pull_operations=%r, stats_enabled=%r, recorders=%s)" % \
-               (self.__class__.__name__, self.url, creds_repr, self.conn_id,
-                self.default_namespace, x509_repr, self.verify_callback,
-                self.ca_certs, self.no_verification, self.timeout,
-                self.use_pull_operations, self.stats_enabled,
+               (self.__class__.__name__, self.url, creds_repr,
+                self.conn_id, self.default_namespace, x509_repr,
+                self.verify_callback, self.ca_certs, self.no_verification,
+                self.timeout, self.use_pull_operations, self.stats_enabled,
                 recorder_list)
 
+    @classmethod
+    def configure_logger(cls, name, log_dest=DEFAULT_LOG_DESTINATION,
+                         detail_level=None,
+                         log_filename=DEFAULT_LOG_DESTINATION,
+                         connection=None):
+        """
+        A log configuration method to configure a python logger to
+        log user-issued calls to :class:`~pywbem.WBEMConnection` methods that
+        drive WBEM operations (see :ref:`WBEM operations`) and the
+        responses/exceptions before they are passed back to the user.
+
+        Parameters:
+
+          name (:term:`string`):
+            Simple name (ex. 'api') for the log to be configured or `all` if
+            all of the :class:`~pywbem.WBEMConnection` loggers are to
+            be configured. The value must be one the named defined in
+            :data:`~pywbem._logging.LOGGER_SIMPLE_NAMES`
+
+          log_dest (:term:`string`):
+            String defining the destination for this log or None.
+
+            If it is a :term:`string` defining one of the destinations defined
+            in :data:`~pywbem._logging.LOG_DESTINATIONS` a Python logger is
+            configured for that destination including configuring a logging
+            handler, the message format and setting the log level to DEBUG.
+
+            If `None` a logger with the Python Python NullHandler is configured.
+
+          detail_level (:term:`string`, :class:`int` or None):
+            Defines the level of detail contained in each log record created.
+
+            If if is a :term:`string` it must be one of the key words that
+            defines the level of detail to be output defined in
+            :data:`~pywbem._logging.LOG_DETAIL_LEVELS`.
+
+            If  it is a :class: `int` it defines the maximum size of the log
+            record record created.
+
+            If `None` it is set to the default detail level for that logger.
+
+          log_filename (:term:`string`):
+            Filename to use as logging file if the log destination is `file`.
+            Ignored if log destination is not `file`. If value is `None` and
+            dest = `file` log, ValueError is raised.
```

Comment:
Instead of the last sentence, we can simply say that it is required to be a file if dest is file.

Reply:
DONE